runtime/vm/ObjectFieldInfo.cpp

```diff
@@ -44,6 +45,26 @@ ObjectFieldInfo::countInstanceFields(void)
 		const U_32 modifiers = field->modifiers;
 		if (J9_ARE_NO_BITS_SET(modifiers, J9AccStatic) ) {
 
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+			J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(field);
+			U_8 *fieldSigBytes = J9UTF8_DATA(J9ROMFIELDSHAPE_SIGNATURE(field));
+			if ('Q' == *fieldSigBytes) {
```

Comment:
Why can modifiers not be used here?

Reply:
I realize now -  but can this be nested inside the object case below? I think making Q behaviour default to L behaviour unless otherwise directed makes sense.